src/zarr/abc/store.py

```diff
@@ -412,6 +412,18 @@ async def getsize_prefix(self, prefix: str) -> int:
         sizes = await concurrent_map(keys, self.getsize, limit=limit)
         return sum(sizes)
 
+    def _as_immutable(self: Self) -> Self:
```

Comment:
in icechunk we have these as `set_read_only` and `set_writeable` which I mildly prefer. Simple words are nice.